File: Lesson13/retrofitapp/src/main/java/ru/mirea/gupalodr/retrofitapp/MainActivity.java
package ru.mirea.gupalodr.retrofitapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class MainActivity extends AppCompatActivity {
    public static final String TAG = "MainActivity";
    public static final String BASE_URL = "https://jsonplaceholder.typicode.com/";
    private ApiService apiService;
    private RecyclerView recyclerView;
    private TodoAdapter todoAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
        apiService = retrofit.create(ApiService.class);

        //creating an api call
        Call<List<Todo>> call = apiService.getTodos();

        //making the call
        call.enqueue(new Callback<List<Todo>>() {
            @Override
            public void onResponse(Call<List<Todo>> call, Response<List<Todo>> response) {
                if (response.isSuccessful() && response.body() != null) {
                    List<Todo> todos = response.body();
                    todoAdapter = new TodoAdapter(MainActivity.this, todos);
                    recyclerView.setAdapter(todoAdapter);
                } else {
                    Log.e(TAG, "onResponse: " + response.code());
                }
            }

            @Override
            public void onFailure(Call<List<Todo>> call, Throwable throwable) {
                Toast.makeText(getApplicationContext(), throwable.getMessage(),
                        Toast.LENGTH_LONG).show();
                Log.e(TAG, "onFailure: " + throwable.getMessage());
            }
        });
    }
}

File: Lesson2/toastapp/src/main/java/ru/mirea/gupalodr/toastapp/MainActivity.kt
package ru.mirea.gupalodr.toastapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    private lateinit var letterCount: Button
    private lateinit var inputText: EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        letterCount = findViewById(R.id.letterCount)
        inputText = findViewById(R.id.inputText)
        
        letterCount.setOnClickListener {
            val count = inputText.text.length
            Toast.makeText(
                this,
                "СТУДЕНТ № 6\nГРУППА БСБО-10-21\nКоличество символов - $count",
                Toast.LENGTH_LONG
            ).show()
        }
    }
}